providers/fips/fipsprov.c

```diff
@@ -605,6 +622,39 @@ static const OSSL_ALGORITHM *fips_query(void *provctx, int operation_id,
     return NULL;
 }
 
+static const OSSL_ALGORITHM *fips_query_internal(void *provctx, int operation_id,
+                                                 int *no_cache)
+{
+    *no_cache = 0;
+
+    if (!ossl_prov_is_running())
+        return NULL;
+
+    switch (operation_id) {
+    case OSSL_OP_DIGEST:
+        return fips_digests;
+    case OSSL_OP_CIPHER:
+        return exported_fips_ciphers;
+    case OSSL_OP_MAC:
+        return fips_macs_internal;
+    case OSSL_OP_KDF:
+        return fips_kdfs;
+    case OSSL_OP_RAND:
+        return fips_rands;
+    case OSSL_OP_KEYMGMT:
+        return fips_keymgmt;
+    case OSSL_OP_KEYEXCH:
+        return fips_keyexch;
+    case OSSL_OP_SIGNATURE:
+        return fips_signature;
+    case OSSL_OP_ASYM_CIPHER:
+        return fips_asym_cipher;
+    case OSSL_OP_KEM:
+        return fips_asym_kem;
```

Comment:
Is it worth trimming this down to just the ones that are required?
Digest, cipher, MAC and rands at a guess.

Reply:
Dont think it really matters..

Reply:
Not a lot I guess.  I was thinking that it would highlight which algorithms are internally fetched (& thus could have an _internal_ flag meaningfully added).

Reply:
We have to be careful with this. It is useful when the higher level needs to do something different to the base,
not sure doing it for digest, ciphers is a good idea if you have to reproduce the checks in the higher level everywhere.